packages/hub/searchers.js

```diff
@@ -86,7 +86,7 @@ class Searchers {
       });
       let pristineResult = await documentContext.pristineDoc();
       if (pristineResult) {
-        authorizedResult = await schema.applyReadAuthorization(pristineResult, { session, type, id });
+        authorizedResult = await documentContext.applyReadAuthorization(pristineResult, { session, type, id });
```

Comment:
@balinterdi same question around applying document context to an aritrary document vs the pristine document already contained in the document context. I think we are abusing document context by allowing it apply read authorization against documents that are not part of its context.